### PR TITLE
Add label fields to merge operation

### PR DESCRIPTION
**Pull request by adr1ancho**
### JIRA link (if applicable) ###

RDM-2323

### Change description ###

Add label fields to merge operation

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ccd/domain/model/search/SearchResultViewItem.java [hunk 1]
```diff
@@ -16,9 +19,13 @@ public SearchResultViewItem() {
...
 
     public SearchResultViewItem(final String caseId,
-                                final Map<String, JsonNode> caseFields) {
+                                final Map<String, JsonNode> caseFields,
+                                final List<CaseField> labelFields) {
         this.caseId = caseId;
         this.caseFields = caseFields;
+        labelFields
+            .stream()
+            .forEach(f -> caseFields.put(f.getId(), JsonNodeFactory.instance.textNode(f.getLabel())));
```
Comment (vlaurin) [inline src/main/java/uk/gov/hmcts/ccd/domain/model/search/SearchResultViewItem.java]: The tests asserting the logic above are missing. Coverage is earned from other tests that are not directly related.

File: src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/MergeDataToSearchResultOperation.java [hunk 2]
```diff
@@ -36,19 +37,24 @@ public SearchResultView execute(final CaseType caseType, final List<CaseDetails>
...
+        final SearchResultViewItem[]
+            viewItems =
+            caseDetails.stream()
+                       .map(caseData -> new SearchResultViewItem(caseData.getReference().toString(),
+                                                                 caseData.getData(),
+                                                                 caseType.getCaseFields()
+                                                                         .stream()
+                                                                         .filter(caseField -> LABEL_FIELD_TYPE.equals(
+                                                                             caseField.getFieldType().getId()))
+                                                                         .collect(Collectors.toList())))
```
Comment (vlaurin) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/MergeDataToSearchResultOperation.java]: Similarly, there's no tests asserting this logic, coverage is inherited from other tests

File: src/main/java/uk/gov/hmcts/ccd/domain/model/search/SearchResultViewItem.java (hunk 1 above)
Comment (adr1ancho) [reply src/main/java/uk/gov/hmcts/ccd/domain/model/search/SearchResultViewItem.java]: Added SearchResultViewItemTest

File: src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/MergeDataToSearchResultOperation.java (hunk 2 above)
Comment (adr1ancho) [reply src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/MergeDataToSearchResultOperation.java]: Added test coverage https://github.com/hmcts/ccd-data-store-api/pull/132/files#diff-4b18d40268e75140ca920fc5cc9dab36

File: src/main/java/uk/gov/hmcts/ccd/domain/model/search/SearchResultViewItem.java (hunk 1 above)
Comment (vlaurin) [reply src/main/java/uk/gov/hmcts/ccd/domain/model/search/SearchResultViewItem.java]: 👍

File: src/main/java/uk/gov/hmcts/ccd/domain/model/search/SearchResultViewItem.java [hunk 3]
```diff
@@ -10,14 +13,19 @@
     @JsonProperty("case_fields")
     private Map<String, Object> caseFields;
 
+    /** Default constructor for Json Mapper. */
     public SearchResultViewItem() {
         // Default constructor for JSON mapper
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/domain/model/search/SearchResultViewItem.java]: can we delete this line?

File: src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/MergeDataToSearchResultOperation.java [hunk 4]
```diff
@@ -38,19 +40,24 @@ public SearchResultView execute(final CaseType caseType, final List<CaseDetails>
...
 
-        final List<SearchResultViewItem> viewItems = caseDetails.stream()
-            .map(caseData -> new SearchResultViewItem(caseData.getReference().toString(), caseData.getCaseDataAndMetadata()))
-            .collect(Collectors.toList());
+        final List<SearchResultViewItem>
+            viewItems =
+            caseDetails.stream()
+                       .map(caseData -> new SearchResultViewItem(caseData.getReference().toString(),
+                                                                 caseData.getCaseDataAndMetadata(),
+                                                                 caseType.getCaseFields()
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/MergeDataToSearchResultOperation.java]: can we simplify this statement? feels too complex to me. One way would be to move 
```
caseType.getCaseFields() .stream()
                                                                         .filter(caseField -> LABEL_FIELD_TYPE.equals(
                                                                             caseField.getFieldType().getType()))
                                                                         .collect(Collectors.toList())))

                                                                        
to a method in CaseType called `getFieldsByType(String type)`

File: src/main/java/uk/gov/hmcts/ccd/domain/model/search/SearchResultViewItem.java (hunk 1 above)
Comment (mario-paniccia) [reply src/main/java/uk/gov/hmcts/ccd/domain/model/search/SearchResultViewItem.java]: I think this logic belongs to MergeDataToSearchResultOperation. 
we should try to keep View type objects as simple as possible, ideally just plain data holders
